src/Compilers/Core/Portable/Diagnostic/SourceLocation.cs

```diff
@@ -108,7 +108,7 @@ public bool Equals(SourceLocation other)
                 return true;
             }
 
-            return other != null && other._syntaxTree == _syntaxTree && other._span == _span;
+            return other is { _syntaxTree: _syntaxTree, _span: _span };
```

Comment:
>_syntaxTree [](start = 43, length = 11)

Not a valid pattern.

Reply:
➡️ Treating this one as a duplicate of https://github.com/dotnet/roslyn/pull/39738#discussion_r344416577